### DISCUSSION TITLE
Sugesstion "don't force" or "bypass" if don't have the blocking rules and allow rule on dnscrypt

### DISCUSSION THREAD

**Discussion by t0ny54** [DNSCrypt/dnscrypt-proxy#2041] [Ideas]
Because accidentally deleting the IP blocking rules txt, dnscrypt proxy service stop working.
My scheduller task failed to complete batch script and make accidently that case after restart dnscrypt proxy...

I suggest only give warning or hint if don't have text of IP blocking rules, blocking rules, allowed names, etc insteded than stop dnscrypt proxy.
That warning can show in dnscrypt log and make log default enable...

Simple sugesstion "don't force" or "bypass" if that blocking rule or allow rule don't have *.txt file and make dnscrypt proxy keep work without that file...

Thank you

**Comment by jedisct1** [comment]
Ignoring configuration parameters if the configuration file is invalid doesn't seem like a good idea.

Since this is a script automatically creating that configuration, maybe you can run `dnscrypt-proxy -show-certs` on the resulting configuration, and only deploy it if the command exits without an error? That way, it will catch any invalid configuration.

**Comment by t0ny54** [comment]
Alternative is that the dnscrypt proxy creates a sqlite3 database and updates it periodically.
Even if there is no such file, can use the old configuration and still work.
Gives a warning in the log and the user can fix it...
The important thing is make dnscrypt keep running...

But I'm not a programmer and maybe that's really hard to do it 😭

"dnscrypt-proxy -show-certs" i will try...
and adjust/edit some batch script in task scheduler script to make that accident not repeated again...
Thank you
